### ISSUE TITLE
wrong report corresponding to function-naming in Test

### ISSUE THREAD

**Issue by soudai-s** [pinterest/ktlint#2508]
A function naming violation is reported in the test file, is this behavior correct?
Apparently it didn't happen with v0.47.1.

## Expected Behavior
Nothing.

## Observed Behavior
Be Reported as function-naming.

## Steps to Reproduce

```
package com.example.presentation.viewmodel

import androidx.test.ext.junit.runners.AndroidJUnit4
import kotlinx.serialization.ExperimentalSerializationApi
import org.junit.Test
import org.junit.runner.RunWith
// and more packages...

@ExperimentalSerializationApi
@RunWith(AndroidJUnit4::class)
class ViewModelTest {
  @Test
  fun fetch_条件1() {
    // do something
  }
}

↑ this example is reported by ktlint as function-naming violation, is it correct?

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version of ktlint used: 1.0.0 and also the same as 1.1.1
* Relevant parts of the `.editorconfig` settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):
* Version of Gradle used (if applicable):
* Operating System and version: Mac OS

**Comment by paul-dingemans** [comment]
> this example is reported by ktlint as function-naming violation, is it correct?

The rule is based on https://kotlinlang.org/docs/coding-conventions.html#function-names. This guideline has been loosened so that underscores are allowed in names of test cases. Also diacritics and stokes on letters (for example `è` or `ŷ`) are allowed.

Characters like `条件` do not comply with this rule. In your case it might be best to [disable this entire rule](https://pinterest.github.io/ktlint/latest/faq/#how-do-i-enable-or-disable-a-rule) or to start using [function names wrapped between backticks](https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods).

**Comment by soudai-s** [comment]
Could you change the behavior for multi-byte characters of function-naming in test similar to the snake case?
I think the base rule https://kotlinlang.org/docs/coding-conventions.html#function-names does apply to the code except test, since ktlint permit the snake case in the test.

**Comment by paul-dingemans** [comment]
> Could you change the behavior for multi-byte characters of function-naming in test similar to the snake case?
> I think the base rule https://kotlinlang.org/docs/coding-conventions.html#function-names does apply to the code except test, since ktlint permit the snake case in the test.

Based on references below, I do not agree with this.

https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods

> In tests (and only in tests), you can use method names with spaces enclosed in backticks. Note that such method names are currently not supported by the Android runtime. Underscores in method names are also allowed in test code.

https://developer.android.com/kotlin/style-guide#naming_2

> Identifiers use only ASCII letters and digits, and, in a small number of cases noted below, underscores. Thus each valid identifier name is matched by the regular expression \w+.
> 
> Special prefixes or suffixes, like those seen in the examples name_, mName, s_name, and kName, are not used except in the case of backing properties (see [Backing properties](https://developer.android.com/kotlin/style-guide#backing-properties)).

**Comment by soudai-s** [comment]
Ok, Thank you for the consideration.
